tests/unit/amo/components/TestAddon.js

```diff
@@ -96,6 +123,72 @@ describe('Addon', () => {
     expect(rootNode.querySelector('h1').textContent).toContain('Chill Out');
   });
 
+  it('renders without an add-on', () => {
+    const errorHandler = new ErrorHandler({
+      id: 'no-addon-error-handler',
+      dispatch: sinon.stub(),
+    });
+    const slugParam = 'some-addon'; // as passed through the URL.
+    const fakeDispatch = sinon.stub();
+
+    // Simulate the case when an add-on has not been loaded into state yet.
+    const root = shallowRender({
+      addon: null,
+      errorHandler,
+      dispatch: fakeDispatch,
+      params: { slug: slugParam },
+    });
+
+    // Since there's no add-on, it should be fetched on load.
+    sinon.assert.calledWith(
+      fakeDispatch, fetchAddonAction({ errorHandler, slug: slugParam }));
+
+    // These should be empty:
+    expect(root.find(InstallButton)).toHaveLength(0);
+    expect(root.find(AddonCompatibilityError)).toHaveLength(0);
+    expect(root.find(AddonMoreInfo)).toHaveLength(0);
+    expect(root.find(RatingManager)).toHaveLength(0);
+    expect(root.find('.AddonDescription-contents')).toHaveLength(0);
+
+    // These should show LoadingText
+    expect(root.find('.Addon-summary').find(LoadingText)).toHaveLength(1);
+    expect(root.find('.Addon-title').find(LoadingText)).toHaveLength(1);
+    expect(root.find('.Addon-metadata').find(LoadingText)).toHaveLength(1);
+  });
+
+  it('renders an error if there is one', () => {
+    const errorHandler = new ErrorHandler({
+      capturedError: new Error('some error'),
+      id: 'some-handler',
+      dispatch: sinon.stub(),
+    });
+
+    const root = shallowRender({ errorHandler });
+    expect(root.find(ErrorList)).toHaveLength(1);
+  });
+
+  it('renders 404 page for missing add-on', () => {
+    const id = 'error-handler-id';
+    const store = createStore().store;
```

Comment:
Could use `const { store } = createStore();` here... and I'd argue we should try to just use `dispatchClientMetadata()` over raw `createStore()` wherever possible because it loads data we can use if we ever have to and is just as easy to use.

But up to you on the usage of `createStore()` vs the dispatch helpers. Obviously these tests are mostly not dispatching things so it doesn't really matter.